PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,334 @@
+            are created when you use the -match operator. This isn't as flexible nor as robust as using [regex]::Matches(), but it's handy
+            in a pinch.
+            
+            In this case, you'll see that the "sh" in "shears" weren't part of the match. The pattern "zero or more p's, followed by e a r
+            s" is found in the string, even though there are parts of the string that aren't relevant to that matching effort.
+
+            $matches[0] is always your most recent match.
+        #>
+
+        $matches[0] | should -Be '__'
+    }
+
+    # +
+    it 'specifies 1 or more of something' {
+        # The + (plus) symbol is a lot like the star, but instead of matching zero or more, it matches one or more.
+
+        '__' -match 'p+ickles' | should -BeTrue
+
+        $mightBePumpkin = '__'
+        $mightBePumpkin += 'umpkin'
+        $mightBePumpkin -match 'p+umpkin' | should -BeFalse
+    }
+
+    # ?
+    it 'specifies 0 or 1 of something' {
+        # The ? (question mark) matches "zero or one" of something.
+        
+        [bool]$boatMatch = __ # Should be either $true or $false
+        'flying through the sky' -match 'f+lying' | should -Be $boatMatch
+        
+        [bool]$floatMatch = __ # Should be either $true or $false
+        'floating away' -match 'b+oat' | should -Be $floatMatch
+    }
+}
+
+describe 'Special Symbols' {
+    <#
+        In regex, there are far more symbols with unique meanings than just quantifiers. What comes next is not a definitive guide to EVERY
+        SINGLE ONE, but rather an introduction to some of the most common and useful special symbols in regex.
+        
+        This is ^\where things st\Art getting \weir\d$
+    #>
+    
+
+    # . (period)
+    it 'matches any character' {
+        # The . (period) matches literally any character
+
+        [bool]$morningMatch = __ # Should be either $true or $false
+        'Lazy Sunday mornings' -match 'S.nday' | should -Be $morningMatch
+
+        '__' -match 'invi.a.ion' | should -BeTrue
+    }
+
+    # \n
+    it 'matches new lines' {
+        <#
+            The \n symbol is the first regex symbol you've encountered that isn't just a single character. Normally if you saw a lowercase "n" in
+            a regex, it would just mean "literally the letter n". When it's preceded by a backslash, however, that "n" takes on a special meaning.
+            In this case, \n matches new lines.
+
+            There are many more regex symbols that are single letters preceded by a backslash. In fact, the backslash is probably the single most
+            important character in all of regex.
+        #>
+
+        $multiLine = @"
+They might look
+similar and look
+soft, but you should
+never confuse a
+wild cougar for
+a domestic cat
+"@
+        $catPattern = '__'
+        $catPattern += '\n'
+        $multiLine -match $catPattern | should -BeTrue
+    }
+
+    # \d \D
+    it 'matches digits and non-digits' {
```

Comment:
I don't know. I like the idea of keeping them together, so that people are aware from the start and don't get tripped up by thinking `\d` and `\D` are the same because they're both the same letter.

Reply:
Fair enough! Even so, each of these sections could stand to be `Context` blocks in their own right.